platform/security/platform-security-core-api/src/main/java/ddf/security/principal/GuestPrincipal.java

```diff
@@ -65,9 +68,25 @@ public static String parseAddressFromName(String fullName) {
         if (!StringUtils.isEmpty(fullName)) {
             String[] parts = fullName.split(NAME_DELIMITER);
             if (parts.length == 2) {
-                return parts[1];
+                return formatIpAddress(parts[1]);
             }
         }
         return null;
     }
+
+    //IPv6 addresses should be contained within brackets to conform
+    //to the spec IETF RFC 2732
+    private static String formatIpAddress(String ipAddress) {
+        try {
+            if (InetAddress.getByName(ipAddress) instanceof Inet6Address) {
```

Comment:
Could you just use StringUtils.prependIfMissing() and StringUtils.appendIfMissing() here?
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#prependIfMissing(java.lang.String,%20java.lang.CharSequence,%20java.lang.CharSequence...)
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#appendIfMissing(java.lang.String,%20java.lang.CharSequence,%20java.lang.CharSequence...)